lib/workers/repository/process/index.ts

```diff
@@ -71,6 +71,35 @@ async function getBaseBranchConfig(
     // baseBranches value should be based off the default branch
     baseBranchConfig.baseBranches = config.baseBranches;
   }
+  else if (
+    config.useBaseBranchConfig === 'replace' &&
+    baseBranch !== config.defaultBranch
+  ) {
+    logger.debug(
+      { baseBranch },
+      `Replacing config from base branch because useBaseBranchConfig=replace`
+    );
+
+    const branchConfigFileName = 'TBD'; // TODO: re-detect config file name on `baseBranch`
```

Comment:
i would prefer to enforce same file name, as it makes everything simpler. anything else can be done in separate follow up PR.

Reply:
I understand that it'd be the most simple solution to rely on the already detected filename from the default branch, but I've gathered my thoughts about this limitation on this discussion comment: https://github.com/renovatebot/renovate/discussions/22264#discussioncomment-6111232

if you guys have the time, please read through the rationale behind this "replace" logic running totally independently from the default branch and its configuration

tl;dr - enforcing the same config file name might be an easy solution, but defeats half the purpose of the whole change, meaning to be able to fully test changes in `renovate.json` (and other config files) on side branches, independently of the "stable" configuration running (most probably, but not necessarily) on the default branch

Reply:
let's do the detection inside a separate PR, so this can be merged faster.